### ISSUE TITLE
Issue: Absolute TypeScript Imports without Paths not working

### ISSUE THREAD

**Issue by fabb** [sverweij/dependency-cruiser#426]
I'm using absolute import paths like `import { X } from 'src/x'` in a next.js project. In the tsconfig.json, `"baseUrl": "."` is set, as [described in next.js documentation](https://nextjs.org/blog/next-9-4#absolute-imports-and-aliases). Next.js compiles fine, and so do builds with `tsc`, `babel`, `webpack`, `jest` and `eslint` in that project.

### Expected Behavior

Dependency cruiser should be able to resolve these absolute import paths.

### Current Behavior

Several rules fail:
* `not-to-unresolvable` (`"to": {"couldNotResolve": true`})
* `no-unreachable-from-root` (`"to": {"reachable": false`})
* `no-orphans` (`"from": {"orphan": true`})

### Possible Solution

I found a workaround that seems not to affect the other tools: in tsconfig.json, set this path alias:
```
"paths": { "*": ["./*"] },
```

Of course it would be nicer if dependency cruiser would not need this.

### Steps to Reproduce (for bugs)

https://codesandbox.io/s/dependency-cruiser-absolute-imports-bug-29u6t?file=/package.json

1. In the sandbox "Terminal" tab, open a new bash terminal with the `+`
2. Execute `npm run depcruise`

-- or --

https://github.com/fabb/dependency-cruiser-absolute-imports-bug

1. Execute `npm i && npm run depcruise`

### Context

The tsconfig file is correctly configured for dependency cruiser:
```
"tsConfig": {
    "fileName": "./tsconfig.json"
}
```

### Your Environment

- Version used: 9.21.6
- Node version: v14.15.4
- Operating System and version: macOS Big Sur
- next.js version: 10.0.5
- Link to your project: private

**Comment by sverweij** [comment]
Hello @fabb thanks for raising this issue and for the reproduction repository (+ code sandbox - never used it before, but it works nicely!).

This bug got introduced with #419 - which limited loading tsconfig-paths-plugin to tsconfigs that actually have at least one `paths` defined as a performance improvement. Apparently tsconfig-paths has a side effect where it adds a resolution rule to check against the baseUrl. You're the second to bump into this, so I've rolled back #419 until there is a more satisfying solution.

For now released as a beta (`dependency-cruiser@9.21.7-beta-1`). I've also ran it in a [fork of the code sandbox](https://codesandbox.io/s/dependency-cruiser-absolute-imports-bug-forked-umylo?file=/.dependency-cruiser.json) with the beta, and it seems to solve the issue.

=> Can you confirm?

With 9.21.6:
```sh
> depcruise --validate .dependency-cruiser.json src pages


  error not-to-unresolvable: pages/index.tsx → src/X
  error not-to-unresolvable: pages/index.tsx → ✖
  error no-unreachable-from-root: src/X.tsx
  error no-orphans: src/X.tsx

✖ 4 dependency violations (4 errors, 0 warnings). 4 modules, 2 dependencies cruised.
```

With 9.21.7-beta-1:
```sh
> depcruise --validate .dependency-cruiser.json src pages

  error not-to-unresolvable: pages/index.tsx → ✖

✖ 1 dependency violations (1 errors, 0 warnings). 3 modules, 2 dependencies cruised.
```

> The remaining not-to-unresolvable has another root cause - under some circumstances the `import` keyword within a 
> jsx/ tsx tag (like here _<h1>This shows absolute import errors with dependency-cruiser</h1>_) will trip up something
> in the transpilation train (related to #395 - my guess is that when transpiring from .tsx to .jsx dependency-cruiser forgets
> to tell the javascript compiler that it's parsing jsX and not plain javascript, I'll look into that separately). 
>
> If you're OK with having pre-compilation dependencies show up as well it can be worked around by using the option
> `"tsPreCompilationDeps": true` as that directly uses the AST produced by typescript (which makes dependency-cruiser
> run about 1.6x as fast to boot).


With 9.21.7-beta-1 and either tsPrecompilationDeps: true or the `imports` string in the tsx's <h1> tag modified:

```sh
> depcruise --validate .dependency-cruiser.json src pages


✔ no dependency violations found (2 modules, 1 dependencies cruised)
```

**Comment by fabb** [comment]
Thanks for the quick response! I tried `dependency-cruiser@9.21.7-beta-1` in my project with the `paths` workaround removed, and it works correctly as expected 👌

**Comment by sverweij** [comment]
Nice, thanks! I'll merge it and put it in the next next regular release (ETA: tonight (GMT+1)).

**Comment by sverweij** [comment]
It's merged and published with `dependency-cruiser@9.21.7`
